File: plugins/csharp-kit/testdata/bad/Smell.cs
using System;
using System.Data.SqlClient;
using System.Threading.Tasks;

namespace Demo;

public class Smell
{
    public void LogBad()
    {
        Console.WriteLine("bad");
    }

    public SqlDataReader SqlBad(SqlConnection conn, string id)
    {
        var cmd = new SqlCommand("SELECT * FROM users WHERE id = " + id, conn);
        return cmd.ExecuteReader();
    }

    public int BlockBad(Task<int> task)
    {
        return task.Result;
    }
}

File: plugins/java-kit/testdata/bad/Smell.java
package demo;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;

public class Smell {
  public void logBad() {
    System.out.println("bad");
  }

  public void stackBad(Exception ex) {
    ex.printStackTrace();
  }

  public ResultSet sqlBad(Connection conn, String id) throws Exception {
    Statement st = conn.createStatement();
    return st.executeQuery("SELECT * FROM users WHERE id = " + id);
  }

  public int npeBad(String s) {
    try {
      return s.length();
    }
    catch (NullPointerException ex) {
      return 0;
    }
  }
}
